### PR TITLE
[sdk] update to GSDK 4.1.2

### PR DESCRIPTION
**Pull request by lmnotran**
This updates the `gecko_sdk` submodule to `4.1.2` and also updates the PAL in `src/src`

- Update to Gecko SDK 4.1.2

### CODE REVIEW

File: src/src/openthread-core-efr32-config.h [hunk 1]
```diff
@@ -47,19 +47,20 @@
 #endif
 
 #include "board_config.h"
+#include "em_device.h"
```
Comment (bertoldi-silabs) [inline src/src/openthread-core-efr32-config.h:50]: It looks like this included file on [GSDK v4.1.2](https://stash.silabs.com/projects/GSDK/repos/gsdk/browse/protocol/openthread/platform-abstraction/efr32/openthread-core-efr32-config.h?at=v4.1.2#50) is `"em_msc.h"`. Does including `em_device.h` accomplish the same thing, and/or was it an intentional replacement?

Comment (lmnotran) [reply src/src/openthread-core-efr32-config.h:50]: Yes, this is intentional. `em_msc.h` is the header file for our `Memory System Controller`. We don't need access to the definitions and types in `em_msc.h` across the entire openthread project, just in [flash.c](https://github.com/openthread/ot-efr32/blob/main/src/src/flash.c#L75-L92).

What we do need access to is the device-specific defines, etc provided by `em_device.h`, which `em_msc.h` does include, but including `em_device.h` directly is more correct.

**Review by romacdon — APPROVED** [review]
LGTM

**Review by bertoldi-silabs — APPROVED** [review]

**Review by parag-silabs — APPROVED** [review]

**Review by hemanth-silabs — APPROVED** [review]
LGTM 👍

**Review by suveshpratapa — APPROVED** [review]
LGTM

**Comment by lmnotran** [comment]
@jwhui could you review when you get a chance please?

**Review by jwhui — APPROVED** [review]

**Pull request merged**